### PR TITLE
fix nil pointer, see 1198

### LINKED ISSUE #1198: nil pointer on Unmarshal Tx
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd0 pc=0x2c288cd]

goroutine 12816418 [running]:
go.vocdoni.io/dvote/vochain/transaction/vochaintx.(*Tx).Unmarshal(0xc01c6335e0, {0x0, 0x0, 0x0}, {0xc002930060, 0xe})
	go.vocdoni.io/dvote/vochain/transaction/vochaintx/vochaintx.go:39 +0x20d
go.vocdoni.io/dvote/vochain.(*BaseApplication).CheckTx(0xc0044d6420, {0x40?, 0x36c0a00?}, 0xc01c62c4a0)
	go.vocdoni.io/dvote/vochain/cometbft.go:196 +0x44a
github.com/cometbft/cometbft/abci/client.(*localClient).CheckTxAsync(0xc0058107e0, {0x4513f80?, 0x5ff4ba0?}, 0xc01c62c4a0)
	github.com/cometbft/cometbft@v0.38.0/abci/client/local_client.go:51 +0xdf
github.com/cometbft/cometbft/proxy.(*appConnMempool).CheckTxAsync(0xc0055cf710, {0x4513f80, 0x5ff4ba0}, 0x60?)
	github.com/cometbft/cometbft@v0.38.0/proxy/app_conn.go:147 +0x170
github.com/cometbft/cometbft/mempool.(*CListMempool).CheckTx(0xc001b7e300, {0x0, 0x0, 0x0}, 0xc0019886a8, {0xd4c0?, {0x0?, 0x1c93397?}})
	github.com/cometbft/cometbft@v0.38.0/mempool/clist_mempool.go:268 +0x20b
github.com/cometbft/cometbft/rpc/core.(*Environment).BroadcastTxSync(0xc0004b2fc0, 0xc005285100, {0x0, 0x0, 0x0})
	github.com/cometbft/cometbft@v0.38.0/rpc/core/mempool.go:35 +0xec
github.com/cometbft/cometbft/rpc/client/local.(*Local).BroadcastTxSync(...)
	github.com/cometbft/cometbft@v0.38.0/rpc/client/local/local.go:98
go.vocdoni.io/dvote/vochain.(*BaseApplication).SetDefaultMethods.func4({0x0?, 0x0, 0x0?})
	go.vocdoni.io/dvote/vochain/appsetup.go:64 +0x4e
go.vocdoni.io/dvote/vochain.(*BaseApplication).SendTx(0xc001dbadb0?, {0x0?, 0x4538128?, 0xc00e17d4c0?})
	go.vocdoni.io/dvote/vochain/app.go:325 +0x2a
go.vocdoni.io/dvote/api.(*API).sendTx(0xc017b0e300?, {0x0?, 0x200?, 0x35e8620?})
	go.vocdoni.io/dvote/api/helpers.go:48 +0x2b
go.vocdoni.io/dvote/api.(*API).accountSetHandler(0xc00dc2c4d0, 0xc0107f68a0, 0xc00059ccf0?)
	go.vocdoni.io/dvote/api/accounts.go:230 +0x2a6
go.vocdoni.io/dvote/httprouter/apirest.(*API).RegisterMethod.func1({{0x35ed6e0, 0xc0107f68a0}, {0xc14febf1307276f2, 0x7baed980287, 0x5fbd0a0}, {0xc0107f68e0, 0x2, 0x2}, 0xc01c62c480})
	go.vocdoni.io/dvote/httprouter/apirest/apirest.go:229 +0x6b
created by go.vocdoni.io/dvote/httprouter.(*HTTProuter).AddPublicHandler.(*HTTProuter).routerHandler.func1 in goroutine 12816372
	go.vocdoni.io/dvote/httprouter/httprouter.go:283 +0x3c5
```

### PR DESCRIPTION
**Pull request by jordipainan**
closes #1198

### CODE REVIEW

File: api/accounts.go [hunk 1]
```diff
@@ -180,17 +180,29 @@ func (a *API) accountHandler(_ *apirest.APIdata, ctx *httprouter.HTTPContext) er
 //	@Router					/accounts [post]
 func (a *API) accountSetHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
 	req := &AccountSet{}
+	if msg == nil {
```
Comment (p4u) [inline api/accounts.go]: you can join both in a single if statement, the second won't be executed if the first fails.

File: api/accounts.go [hunk 2]
```diff
@@ -180,17 +180,29 @@ func (a *API) accountHandler(_ *apirest.APIdata, ctx *httprouter.HTTPContext) er
...
 	metadataURI, err := func() (string, error) {
 		stx := &models.SignedTx{}
+		if req.TxPayload == nil {
+			return "", ErrUnmarshalingServerProto
+		}
 		if err := proto.Unmarshal(req.TxPayload, stx); err != nil {
 			return "", err
 		}
 		tx := &models.Tx{}
+		if stx.GetTx() == nil {
```
Comment (p4u) [inline api/accounts.go]: Reuse the result of stx.GetTx() please

**Review by p4u — CHANGES_REQUESTED** [review]
Please check this same issue does not happend on other handlers.

**Comment by p4u** [comment]
Btw, the issue seems to be at 

```
go.vocdoni.io/dvote/vochain/transaction/vochaintx.(*Tx).Unmarshal(0xc01c6335e0, {0x0, 0x0, 0x0}, {0xc002930060, 0xe})
	go.vocdoni.io/dvote/vochain/transaction/vochaintx/vochaintx.go:39 +0x20d
go.vocdoni.io/dvote/vochain.(*BaseApplication).CheckTx(0xc0044d6420, {0x40?, 0x36c0a00?}, 0xc01c62c4a0)
	go.vocdoni.io/dvote/vochain/cometbft.go:196 +0x44a
```

So the check must exist in the API layer but also on the vochain transaction handler.

**Comment by jordipainan** [comment]
> Btw, the issue seems to be at
> 
> ```
> go.vocdoni.io/dvote/vochain/transaction/vochaintx.(*Tx).Unmarshal(0xc01c6335e0, {0x0, 0x0, 0x0}, {0xc002930060, 0xe})
> 	go.vocdoni.io/dvote/vochain/transaction/vochaintx/vochaintx.go:39 +0x20d
> go.vocdoni.io/dvote/vochain.(*BaseApplication).CheckTx(0xc0044d6420, {0x40?, 0x36c0a00?}, 0xc01c62c4a0)
> 	go.vocdoni.io/dvote/vochain/cometbft.go:196 +0x44a
> ```
> 
> So the check must exist in the API layer but also on the vochain transaction handler.

added extra checks on comet and to the tx unmarshaling content

File: vochain/cometbft.go [hunk 3]
```diff
@@ -327,6 +336,9 @@ func (app *BaseApplication) PrepareProposal(ctx context.Context,
 	req *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
 	app.prepareProposalLock.Lock()
 	defer app.prepareProposalLock.Unlock()
+	if req == nil {
+		return nil, errors.New("nil request")
```
Comment (p4u) [inline vochain/cometbft.go]: use fmt.Errorf() please

**Review by p4u — APPROVED** [review]

**Pull request merged**